### PR TITLE
(BTS) IT-4339: bts barebone servers

### PR DESCRIPTION
**Pull request by hreinking**

### CODE REVIEW

**Review by jhoblitt — CHANGES_REQUESTED** [review]
Please remove the empty yaml files. They are only going to be confusing.

File: hieradata/node/auxtel-dc01.ls.lsst.org.yaml [hunk 1]
```diff
@@ -0,0 +1,134 @@
+---
+profile::nm::connections:
```
Comment (jhoblitt) [inline hieradata/node/auxtel-dc01.ls.lsst.org.yaml:2]: Please do not duplicate the interface configuration between the cluster and the nodes. Our pattern has been to put as much network configuration into the cluster layer and a node layer is used only when a host needs configuration which deviates from the majority of the cluster.

File: hieradata/node/tel-hw1.ls.lsst.org.yaml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+      [ipv6]
+      method=disabled
+
+
+profile::core::k5login::k5login:
+  "/home/saluser/.k5login":
+    ensure: "present"
+    principals:
+      - "saluser/tel-lt1.tu.lsst.org@LSST.CLOUD"
```
Comment (jhoblitt) [inline hieradata/node/tel-hw1.ls.lsst.org.yaml:124]: This seems wrong.  TTS hosts should not be accessing BTS hosts.

File: hieradata/node/auxtel-archiver.ls.lsst.org.yaml [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    share: "repo"
+    server: "%{facts.fqdn}"
+    atboot: true
+
+profile::core::k5login::k5login:
+  "/home/saluser/.k5login":
+    ensure: "present"
+    principals:
+      - "iip/forwarders.ls.lsst.org@LSST.CLOUD"
+      - "ccs-ipa/auxtel-fp01.ls.lsst.org@LSST.CLOUD"
```
Comment (jhoblitt) [inline hieradata/node/auxtel-archiver.ls.lsst.org.yaml:162]: Please confirm that this is necessary.

**Review by jhoblitt — CHANGES_REQUESTED** [review]

File: hieradata/cluster/lsstcam-ccs/osfamily/RedHat/major/8.yaml [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+      interface-name=enp129s0f1
+
+      [ethernet]
+
+      [ipv4]
+      method=disabled
+
+      [ipv6]
+      method=disabled
+  enp129s0f1.2502:
```
[the remark this answers is not in the document]
Comment (hreinking) [reply hieradata/cluster/lsstcam-ccs/osfamily/RedHat/major/8.yaml:82]: done

File: hieradata/node/auxtel-dc01.ls.lsst.org.yaml (hunk 1 above)
Comment (hreinking) [reply hieradata/node/auxtel-dc01.ls.lsst.org.yaml:2]: Independent node configuration applied.

File: hieradata/node/tel-hw1.ls.lsst.org.yaml (hunk 2 above)
Comment (hreinking) [reply hieradata/node/tel-hw1.ls.lsst.org.yaml]: changed, done

**Review by tony-johnson — APPROVED** [review]
HI, this looks like too many changes for me to do a meaningful review, but I suggest we go ahead and merge it and then start trying to use it and identify any issues that way. 

For me to do much testing I need the DAQ interfaces setup on all of the lsscam-dcNN nodes, which I don't think has happened yet.

**Review by glennmorris — APPROVED** [review]
I think we would want to tweak some CCS stuff, and hopefully would be able to remove the splitting off of separate RHEL7/8 hiera, but I think it's probably best to just merge this for now and then adjust it later.

File: hieradata/node/lsstcam-dc01.ls.lsst.org.yaml [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+      [ethernet]
+
+      [vlan]
+      flags=1
+      id=2502
+      parent=enp129s0f1
+
+      [bridge-port]
+  dds:
```
Comment (jhoblitt) [inline hieradata/node/lsstcam-dc01.ls.lsst.org.yaml:100]: Where dds interfaces requested on the dc01 nodes? I don't see it on the confluence page and non of the other ccs clusters need it.

File: hieradata/node/auxtel-dc01.ls.lsst.org.yaml (hunk 1 above)
Comment (jhoblitt) [reply hieradata/node/auxtel-dc01.ls.lsst.org.yaml:2]: There is a node layer file for all of the lsstcam-* nodes that appear to be identical.

File: hieradata/node/lsstcam-dc01.ls.lsst.org.yaml (hunk 5 above)
Comment (tony-johnson) [reply hieradata/node/lsstcam-dc01.ls.lsst.org.yaml:100]: We do not need dds interfaces on the dcNN nodes. We do need DAQ network interfaces.

Comment (jhoblitt) [reply hieradata/node/lsstcam-dc01.ls.lsst.org.yaml:100]: @tony-johnson I don't think the NICs for the DAQ have been installed. It definitely would have been more efficient to have done that first and then configured all networking at once.

File: hieradata/node/auxtel-dc01.ls.lsst.org.yaml (hunk 1 above)
Comment (hreinking) [reply hieradata/node/auxtel-dc01.ls.lsst.org.yaml:2]: On CP, all lsstcam nodes belong to cluster lsstcam-ccs; you guys want me to separate them? Override them at node level doesn't work for the unit tests. Let'm know.

File: hieradata/node/lsstcam-dc01.ls.lsst.org.yaml (hunk 5 above)
Comment (hreinking) [reply hieradata/node/lsstcam-dc01.ls.lsst.org.yaml:100]: Updating

File: hieradata/node/auxtel-dc01.ls.lsst.org.yaml (hunk 1 above)
Comment (hreinking) [reply hieradata/node/auxtel-dc01.ls.lsst.org.yaml:2]: All dc nodes network configuration were moved into cluster level.

File: hieradata/cluster/lsstcam-ccs/osfamily/RedHat/major/8.yaml [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+      [bridge]
+      stp=false
+
+      [ipv4]
+      method=disabled
+      route1=139.229.153.0/24
+      route1_options=table=2505
+      route2=0.0.0.0/0,139.229.153.254
+      route2_options=table=2505
+      routing-rule1=priority 100 from 193.229.153.64/26 table 2505
```
Comment (jhoblitt) [inline hieradata/cluster/lsstcam-ccs/osfamily/RedHat/major/8.yaml:119]: typo -- looks like the typo was copied from cluster/manke.yaml.

File: hieradata/node/auxtel-archiver.ls.lsst.org.yaml [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+      [bridge]
+      stp=false
+
+      [ipv4]
+      method=disabled
+      route1=139.229.153.0/24
+      route1_options=table=2505
+      route2=0.0.0.0/0,139.229.153.254
+      route2_options=table=2505
+      routing-rule1=priority 100 from 193.229.153.64/26 table 2505
```
Comment (jhoblitt) [inline hieradata/node/auxtel-archiver.ls.lsst.org.yaml:119]: typo

File: hieradata/node/auxtel-fp01.ls.lsst.org.yaml [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+      [bridge]
+      stp=false
+
+      [ipv4]
+      method=disabled
+      route1=139.229.153.0/24
+      route1_options=table=2505
+      route2=0.0.0.0/0,139.229.153.254
+      route2_options=table=2505
+      routing-rule1=priority 100 from 193.229.153.64/26 table 2505
```
Comment (jhoblitt) [inline hieradata/node/auxtel-fp01.ls.lsst.org.yaml:137]: typo

File: hieradata/cluster/lsstcam-ccs/osfamily/RedHat/major/8.yaml [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+      interface-name=enp129s0f1
+
+      [ethernet]
+
+      [ipv4]
+      method=disabled
+
+      [ipv6]
+      method=disabled
+  enp129s0f1.2505:
```
Comment (jhoblitt) [inline hieradata/cluster/lsstcam-ccs/osfamily/RedHat/major/8.yaml:82]: `hieradata/cluster/lsstcam-ccs/osfamily/RedHat/major/8.yaml` doesn't seem like the correct place in the hierarchy for network config as the vlan ids will be different at the summit.

The existing `site/%{site}/cluster/%{cluster}.yaml` would probably be reasonable as there won't ever be EL7 lsstcam-ccs nodes in the `ls` site.

File: site/profile/manifests/archive/common.pp [hunk 10]
```diff
@@ -32,12 +32,14 @@
     ensure_packages($packages)
   }
 
-  class { 'python':
-    version     => 'python36',
-    pip         => 'present',
-    dev         => 'present',
-    venv        => 'present',
-    python_pips => $python_pips,
+  if ((fact('os.release.major') == '7' and fact('os.architecture') == 'x86_64')) {
```
Comment (jhoblitt) [inline site/profile/manifests/archive/common.pp:35]: Please add spec coverage for this change.

File: site/profile/manifests/ccs/graphical.pp [hunk 11]
```diff
@@ -28,9 +28,16 @@
     ## Although people sometimes want to eg use vnc,
     ## so it does end up being needed on servers too.
     ## "Server with GUI" instead? Not much smaller.
-    yum::group { 'GNOME Desktop':
-      ensure  => present,
-      timeout => 1800,
+    # Packages not present on EL8 or EL9
+    if fact('os.release.major') == '7' {
```
Comment (jhoblitt) [inline site/profile/manifests/ccs/graphical.pp:32]: Please add spec coverage for this change.

File: site/profile/manifests/ts/dco.pp [hunk 12]
```diff
@@ -4,6 +4,23 @@
 class profile::ts::dco {
   $user = 'dco'
 
+  file { "/home/${user}/docker_tmp":
```
Comment (jhoblitt) [inline site/profile/manifests/ts/dco.pp:7]: Please add spec coverage for this resource.

File: spec/classes/archive/commmon_spec.rb [hunk 13]
```diff
@@ -19,12 +19,12 @@
         it { is_expected.to contain_package(p) }
       end
 
-      %w[
-        docker-compose
-        cryptography
-      ].each do |p|
-        it { is_expected.to contain_python__pip(p) }
-      end
+      # %w[
```
Comment (jhoblitt) [inline spec/classes/archive/commmon_spec.rb:22]: These pypi packages are still present in `site/profile/data/common.yaml`, why is this block being commented out?  This looks stale as we should generally be using the `docker-compose` rpm now instead of pypi packages.

File: spec/hosts/nodes/tel-hw1.ls.lsst.org_spec.rb [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+
+        it_behaves_like 'nm named interface'
+        it { expect(nm_keyfile['connection']['type']).to eq('bridge') }
+        it { expect(nm_keyfile['connection']['autoconnect']).to be_nil }
+        it { expect(nm_keyfile['bridge']['stp']).to be false }
+        it { expect(nm_keyfile['ipv4']['method']).to eq('auto') }
+        it { expect(nm_keyfile['ipv6']['method']).to eq('disabled') }
+      end
+
+      it { is_expected.to compile.with_all_deps }
```
Comment (jhoblitt) [inline spec/hosts/nodes/tel-hw1.ls.lsst.org_spec.rb:72]: It is conventional to test that the catalog compiles as the first test.

File: hieradata/node/lsstcam-daq-mgt.ls.lsst.org.yaml [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+
+      [ethernet]
+
+      [ipv4]
+      method=disabled
+
+      [ipv6]
+      method=disabled
+
+nfs::server_enabled: true
```
Comment (jhoblitt) [inline hieradata/node/lsstcam-daq-mgt.ls.lsst.org.yaml:83]: Please add spec coverage for the nfs exports.

File: hieradata/node/lsstcam-mcm.ls.lsst.org.yaml [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+
+      [ethernet]
+
+      [vlan]
+      flags=1
+      id=2502
+      parent=enp129s0f1
+
+      [bridge-port]
+  dds:
```
Comment (jhoblitt) [inline hieradata/node/lsstcam-mcm.ls.lsst.org.yaml:100]: Please add spec coverage for the dds interface.

File: hieradata/role/amor.yaml [hunk 17]
```diff
@@ -7,3 +7,4 @@ classes:
   - "profile::core::docker::prune"
   - "profile::core::nfsclient"
   - "profile::ts::dco"
+  - "profile::ts::love"
```
Comment (jhoblitt) [inline hieradata/role/amor.yaml:10]: Please add spec coverage for this change to `spec/hosts/roles/amor_spec.rb`.

File: spec/spec_helper.rb [hunk 18]
```diff
@@ -276,7 +276,6 @@ def node_files
     profile::core::nfsclient
     profile::core::nfsserver
     profile::core::sysctl::lhn
-    python
```
Comment (jhoblitt) [inline spec/spec_helper.rb:279]: As noted elsewhere, there needs to be replacement spec coverage for the installation of python.

**Review by jhoblitt — CHANGES_REQUESTED** [review]
Please remove these empty files:

- hieradata/cluster/amor/osfamily/RedHat/major/8.yaml
- lsst-itconf/hieradata/cluster/azar/osfamily/RedHat/major/8.yaml

Several of the commit messages are not in imperative mood. For example, "auxtel-archive ls node definition" should be "add auxtel-archive ls node definiton". See https://ittn-005.lsst.io/ and https://developer.lsst.io/work/flow.html

There are commits within the PR that don't need to be in the history. For example, these commits changes hieradata/cluster/manke.yaml which are later undone.

3fe3adc951cefb9a3ac87da839fa819ff9e6f493
b8f79763e3cfe69155d1562a44964ec34df245cc

There are a few commit messages that aren't very descriptive such as "fix tests". There are a lot of tests in this repo. Which tests are being fixed?

File: hieradata/node/auxtel-archiver.ls.lsst.org.yaml [hunk 19]
```diff
@@ -0,0 +1,155 @@
...
+nfs::nfs_v4_export_root_clients: >-
+  %{facts.networking.ip}/32(ro,fsid=root,insecure,no_subtree_check,async,root_squash)
+  139.229.151.0/24(ro,fsid=root,insecure,no_subtree_check,async,root_squash)
+  139.229.152.192/26(ro,fsid=root,insecure,no_subtree_check,async,root_squash)
+nfs::nfs_exports_global:
+  /data/lsstdata:
+    clients: >-
+      %{facts.networking.ip}/32(rw,nohide,insecure,no_subtree_check,async,root_squash)
+      139.229.151.0/24(rw,fsid=root,insecure,no_subtree_check,async,root_squash)
+      139.229.152.192/26(rw,nohide,insecure,no_subtree_check,async,root_squash)
```
[the remark this answers is not in the document]
Comment (hreinking) [reply hieradata/node/auxtel-archiver.ls.lsst.org.yaml:134]: Neither of those are the DDS subnet; first one is Manke, and the second one is BTS management (love, azar, lsstcam..)

[the remark this answers is not in the document]
Comment (jhoblitt) [reply hieradata/node/auxtel-archiver.ls.lsst.org.yaml:134]: Indeed. That commit is in error. I'll make it resolved.

File: hieradata/node/auxtel-fp01.ls.lsst.org.yaml [hunk 20]
```diff
@@ -0,0 +1,190 @@
...
+      [ipv6]
+      method=disabled
+
+
+# Server configuration
+nfs::server_enabled: true
+nfs::nfs_v4_export_root_clients: >-
+  %{facts.networking.ip}/32(ro,fsid=root,insecure,no_subtree_check,async,root_squash)
+  139.229.151.0/24(ro,fsid=root,insecure,no_subtree_check,async,root_squash)
+  139.229.152.0/25(ro,fsid=root,insecure,no_subtree_check,async,root_squash)
```
[the remark this answers is not in the document]
Comment (hreinking) [reply hieradata/node/auxtel-fp01.ls.lsst.org.yaml:166]: first one is Manke, and the second one is BTS management

File: hieradata/node/auxtel-dc01.ls.lsst.org.yaml [hunk 21]
```diff
@@ -0,0 +1,134 @@
...
+      [bridge]
+      stp=false
+
+      [ipv4]
+      method=disabled
+      route1=139.229.153.0/24
+      route1_options=table=2505
+      route2=0.0.0.0/0,139.229.153.254
+      route2_options=table=2505
+      routing-rule1=priority 100 from 193.229.153.64/26 table 2505
```
[the remark this answers is not in the document]
Comment (hreinking) [reply hieradata/node/auxtel-dc01.ls.lsst.org.yaml]: fixed

File: hieradata/cluster/lsstcam-ccs/osfamily/RedHat/major/8.yaml (hunk 6 above)
Comment (csilva-cl) [reply hieradata/cluster/lsstcam-ccs/osfamily/RedHat/major/8.yaml]: I think this typo was carried on from another PR, but yes, just a typo.

Comment (hreinking) [reply hieradata/cluster/lsstcam-ccs/osfamily/RedHat/major/8.yaml]: yes, it's fix now

File: hieradata/node/auxtel-archiver.ls.lsst.org.yaml (hunk 7 above)
Comment (hreinking) [reply hieradata/node/auxtel-archiver.ls.lsst.org.yaml]: fixed

File: spec/hosts/nodes/tel-hw1.ls.lsst.org_spec.rb (hunk 14 above)
Comment (hreinking) [reply spec/hosts/nodes/tel-hw1.ls.lsst.org_spec.rb]: changed

File: hieradata/node/lsstcam-mcm.ls.lsst.org.yaml (hunk 16 above)
Comment (hreinking) [reply hieradata/node/lsstcam-mcm.ls.lsst.org.yaml:100]: added.

File: hieradata/node/lsstcam-daq-mgt.ls.lsst.org.yaml (hunk 15 above)
Comment (hreinking) [reply hieradata/node/lsstcam-daq-mgt.ls.lsst.org.yaml:83]: added

File: hieradata/node/lsstcam-mcm.ls.lsst.org.yaml (hunk 16 above)
Comment (hreinking) [reply hieradata/node/lsstcam-mcm.ls.lsst.org.yaml:100]: added

File: hieradata/role/amor.yaml (hunk 17 above)
Comment (hreinking) [reply hieradata/role/amor.yaml:10]: added

File: hieradata/cluster/lsstcam-ccs/osfamily/RedHat/major/8.yaml (hunk 9 above)
Comment (hreinking) [reply hieradata/cluster/lsstcam-ccs/osfamily/RedHat/major/8.yaml:82]: moved

**Review by hreinking — COMMENTED** [review]
fixed

File: hieradata/cluster/lsstcam-ccs/osfamily/RedHat/major/8.yaml (hunk 9 above)
Comment (jhoblitt) [reply hieradata/cluster/lsstcam-ccs/osfamily/RedHat/major/8.yaml:82]: I see that a new file was added that looks like a duplicate of this file but this file remains a concern.

File: site/profile/data/common.yaml [hunk 22]
```diff
@@ -190,13 +190,9 @@ profile::archive::common::packages:
...
   - "git"
+  - "docker-compose-plugin"
 
 profile::archive::common::python_pips:
-  # install docker-compose at system level - "docker"
-  "docker-compose":
-    virtualenv: "system"
-    ensure: "1.25.0"
-    pip_provider: "pip3"
   "cryptography":
```
Comment (jhoblitt) [inline site/profile/data/common.yaml:196]: The `cryptography` package was only needed for `docker-compose` from pypi. It can be removed to.

File: site/profile/manifests/archive/common.pp [hunk 23]
```diff
@@ -32,14 +32,29 @@
     ensure_packages($packages)
   }
 
-  class { 'python':
-    version     => 'python36',
-    pip         => 'present',
-    dev         => 'present',
-    venv        => 'present',
-    python_pips => $python_pips,
+  if fact('os.architecture') == 'x86_64' {
```
Comment (jhoblitt) [inline site/profile/manifests/archive/common.pp:35]: @glennmorris As of https://github.com/lsst-it/puppet-ccs_software/pull/31 , I assume the alternative link is now undesirable? Do you want me to make a new release of `lsst/ccs_software` and bump the version in this PR?

File: spec/classes/archive/commmon_spec.rb [hunk 24]
```diff
@@ -6,6 +6,24 @@
   on_supported_os.each do |os, facts|
     context "on #{os}" do
       let(:facts) { facts }
+      let(:manifest) do
```
Comment (jhoblitt) [inline spec/classes/archive/commmon_spec.rb:9]: Why is this needed?

File: spec/classes/archive/commmon_spec.rb [hunk 25]
```diff
@@ -15,15 +33,13 @@
...
         it { is_expected.to contain_package(p) }
       end
 
-      %w[
-        docker-compose
-        cryptography
-      ].each do |p|
-        it { is_expected.to contain_python__pip(p) }
+      if facts[:os]['release']['major'] == '7' && (facts[:os]['architecture'] == 'x86_64')
+        it { is_expected.to contain_python__pip('cryptography') }
```
Comment (jhoblitt) [inline spec/classes/archive/commmon_spec.rb:42]: As noted on the puppet code, I believe this pip install can be removed.

File: spec/hosts/nodes/lsstcam-mcm.ls.lsst.org_spec.rb [hunk 26]
```diff
@@ -0,0 +1,72 @@
...
+    # rubocop:enable Naming/VariableNumber
+    context "on #{os}" do
+      let(:facts) { override_facts(facts, fqdn: 'lsstcam-mcm.ls.lsst.org') }
+      let(:node_params) do
+        {
+          role: 'ccs-dc',
+          site: 'ls',
+        }
+      end
+      let(:vlan_id) { 2505 }
```
Comment (jhoblitt) [inline spec/hosts/nodes/lsstcam-mcm.ls.lsst.org_spec.rb:18]: Unused and may be removed.

File: spec/hosts/nodes/lsstcam-mcm.ls.lsst.org_spec.rb [hunk 27]
```diff
@@ -0,0 +1,72 @@
...
+      let(:rt_id) { vlan_id }
```
Comment (jhoblitt) [inline spec/hosts/nodes/lsstcam-mcm.ls.lsst.org_spec.rb:19]: Unused and may be removed.

File: hieradata/cluster/manke.yaml [hunk 28]
```diff
@@ -161,7 +161,7 @@ profile::nm::connections:
       route1_options=table=2505
       route2=0.0.0.0/0,139.229.153.254
       route2_options=table=2505
-      routing-rule1=priority 100 from 193.229.153.64/26 table 2505
+      routing-rule1=priority 100 from 139.229.153.64/26 table 2505
```
Comment (jhoblitt) [inline hieradata/cluster/manke.yaml:164]: FYI - https://github.com/lsst-it/lsst-itconf/pull/751 will add spec coverage of this.

**Review by jhoblitt — CHANGES_REQUESTED** [review]

File: site/profile/data/common.yaml (hunk 22 above)
Comment (hreinking) [reply site/profile/data/common.yaml:196]: removed.

File: site/profile/manifests/archive/common.pp (hunk 23 above)
Comment (glennmorris) [reply site/profile/manifests/archive/common.pp:35]: @jhoblitt Yes, I would like a new release of ccs_software please. IMO the version bump could precede this PR, but having it in here is fine too.

File: spec/classes/archive/commmon_spec.rb (hunk 24 above)
Comment (hreinking) [reply spec/classes/archive/commmon_spec.rb:9]: to test
https://github.com/lsst-it/lsst-itconf/blob/IT-4339_bts/site/profile/manifests/archive/common.pp#L35-L57

File: spec/classes/archive/commmon_spec.rb (hunk 25 above)
Comment (hreinking) [reply spec/classes/archive/commmon_spec.rb:42]: removed.

File: spec/hosts/nodes/lsstcam-mcm.ls.lsst.org_spec.rb (hunk 26 above)
Comment (hreinking) [reply spec/hosts/nodes/lsstcam-mcm.ls.lsst.org_spec.rb:18]: removed

File: spec/hosts/nodes/lsstcam-mcm.ls.lsst.org_spec.rb (hunk 27 above)
Comment (hreinking) [reply spec/hosts/nodes/lsstcam-mcm.ls.lsst.org_spec.rb:19]: removed

File: hieradata/cluster/manke.yaml (hunk 28 above)
Comment (hreinking) [reply hieradata/cluster/manke.yaml:164]: you prefer to leave it with the typo?

**Comment by jhoblitt** [comment]
Please let me know when the tests are passing and this PR is ready for re-review.

File: spec/classes/archive/commmon_spec.rb (hunk 24 above)
Comment (jhoblitt) [reply spec/classes/archive/commmon_spec.rb:9]: A manifest block adds resources into the test catalog but it doesn't create a test.  This appears unnecessary and I believe it can be removed.

**Review by jhoblitt — CHANGES_REQUESTED** [review]

**Comment by jhoblitt** [comment]
This PR needs to be rebased against current `production`. There are 4 commits in the branch which are duplicates of commits already merged.

- https://github.com/lsst-it/lsst-control/pull/749/commits/c6617ed8db86c830f77a60c91b4ca3522c3a01ac
- https://github.com/lsst-it/lsst-control/pull/749/commits/50a6996faf9badf13ee50342c54c5da15b8e19f7
- https://github.com/lsst-it/lsst-control/pull/749/commits/f1111ac634dccafc961e70143d389ea82ec97fe9
- https://github.com/lsst-it/lsst-control/pull/749/commits/22500613ad9dd23c2aa513a85246dfdebdf1f7a7

These should go away with the rebase.

**Comment by jhoblitt** [comment]
I'm going to defer looking at the individual commits until the branch has been rebased.
